Composer/packages/server/src/workers/templateInstallation_worker.ts

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { Worker, isMainThread, workerData, parentPort } from 'worker_threads';
+
+import Environment from 'yeoman-environment';
+import yeoman from 'yeoman-environment';
+import TerminalAdapter from 'yeoman-environment/lib/adapter';
+import formatMessage from 'format-message';
+
+import { templateGeneratorPath } from '../settings/env';
+import log from '../logger';
+import { BackgroundProcessManager } from '../services/backgroundProcessManager';
+
+const installRemoteTemplate = async (
+  yeomanEnv: Environment,
+  generatorName: string,
+  npmPackageName: string,
+  templateVersion: string
+): Promise<boolean> => {
+  yeomanEnv.cwd = templateGeneratorPath;
+  try {
+    log('Installing generator', npmPackageName);
+    templateVersion = templateVersion ? templateVersion : '*';
+    await yeomanEnv.installLocalGenerators({ [npmPackageName]: templateVersion });
+
+    log('Looking up local packages');
+    await yeomanEnv.lookupLocalPackages();
+    return true;
+  } catch {
+    return false;
+  }
+};
+
+const instantiateRemoteTemplate = async (
+  yeomanEnv: Environment,
+  generatorName: string,
+  dstDir: string,
+  projectName: string
+): Promise<boolean> => {
+  log('About to instantiate a template!', dstDir, generatorName, projectName);
+  yeomanEnv.cwd = dstDir;
+
+  await yeomanEnv.run([generatorName, projectName], {}, () => {
+    log('Template successfully instantiated', dstDir, generatorName, projectName);
+  });
+  return true;
```

Comment:
Is this return required since the catch handler of the caller will hit if there is an error.

Reply:
good catch, it has been removed